web3/_utils/request.py

```diff
@@ -100,16 +109,22 @@ def json_make_get_request(
 
 
 def get_response_from_post_request(
-    endpoint_uri: URI, *args: Any, **kwargs: Any
+    endpoint_uri: URI,
+    *args: Any,
+    **kwargs: Any,
 ) -> requests.Response:
     kwargs.setdefault("timeout", DEFAULT_TIMEOUT)
-    session = cache_and_return_session(endpoint_uri)
+    provider_id = kwargs.get("provider_id")
```

Comment:
When we pass the `provider_id` along from the `make_request()`, this will likely either need to be a `pop()` or handled in the method signature since it can't be passed on to the `session.post()` call on line 119.

I slightly prefer the explicit expectation / handling in the method signature `provider_id: int = None` but either could work here.